networkx/algorithms/reciprocity.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+#
+#    Copyright (C) 2015 by 
+#    Haochen Wu <wuhaochen42@gmail.com>
+#    All rights reserved.
+#    BSD license.
+"""Algorithms to calculate reciprocity in a directed graph."""
+from networkx import NetworkXError
+from ..utils import not_implemented_for
+
+__author__ = """\n""".join(['Haochen Wu <wuhaochen42@gmail.com>'])
+
+__all__= ['reciprocity','overall_reciprocity']
+
+@not_implemented_for('undirected','multigraph')
+def reciprocity(G, nodes=None):
+    """Computer the reciprocity in a directed graph.
```

Comment:
Computer -> Compute         [line 17 and again in line 64]

Also, after a blank line could you give the definition of reciprocity?

line 25 could be more precise... something like:
If None, return the overall reciprocity of the whole graph.

Finally, it seems friendlier to not raise a NetworkXError for isolated nodes. We usually try to find a way to flag such trouble without halting the rest of the computation.  Might be best to store the reciprocity as None for isolated nodes. What do you think?